Mu2eG4/src/constructDiskCalorimeter.cc

```diff
@@ -289,48 +296,92 @@ namespace mu2e {
        reg.add(rotFPPipe);
        
        G4Torus*         coolFP     = new G4Torus("caloCoolFP",FPCoolPipeRadius-FPCoolPipeThickness, FPCoolPipeRadius, FPCoolPipeTorRadius, angMax, CLHEP::twopi-2.0*angMax);
-       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog_"+std::to_string(idisk),isPipeVisible,G4Color::Red(),isPipeSolid,0);
+       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog"+std::to_string(idisk),isPipeVisible,G4Color::Red(),isPipeSolid,0);
        pv = new G4PVPlacement(rotFPPipe,G4ThreeVector(0.0,0.0,ZposPipe), coolFPLog, "caloCoolFPPV", frontPlateLog, false, 0, false);
        doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);                
        
 
        //pipes nside foam
-       G4RotationMatrix* rotPipe1 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe1->rotateZ(CLHEP::pi/2.0);
-       G4RotationMatrix* rotPipe2 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe2->rotateZ(1.5*CLHEP::pi);
-       G4RotationMatrix* rotPipeFlat = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipeFlat->rotateX(CLHEP::pi/2.0);
-       reg.add(rotPipe1);
-       reg.add(rotPipe2);
-       reg.add(rotPipeFlat);
+       G4RotationMatrix* rotPipe1 = nullptr;
```

Comment:
i think the intend would be clearer if rotPipe1 was declared as a identity rotation. Here you don't know what a nullptr rotation does!

Reply:
> i think the intend would be clearer if rotPipe1 was declared as a identity rotation. Here you don't know what a nullptr rotation does!

Well, the difference between using the identity matrix and nullptr is that the former leads to performing the rotation and all the calculations, which should end where one started, and when one uses the latter the constructor just leaves the input as is, without performing the rotation calculations. 
I agree that using nullptr here may not be very clear, so perhaps one should add a comment as to what it means.
Still better than forcing the identity rotation calculation :).